### PR TITLE
Cleanup documentation

### PR DESCRIPTION
**Pull request by NickNaso**
Start cleaning up documentation like reported here : [https://github.com/nodejs/abi-stable-node/issues/280#issuecomment-402811983](https://github.com/nodejs/abi-stable-node/issues/280#issuecomment-402811983)
I need to do another pass to complete the work, but if anyone want start the review it could be great.

### CODE REVIEW

**Comment by NickNaso** [comment]
Hi everyone I ended to clean up the documentation removing the unnecessary working progress comments and adding the Napi namespace. Please review this work so we can complete the steps to release the first complete version of documentation for `node-addon-api`.

File: doc/async_worker.md [hunk 1]
```diff
@@ -81,13 +81,13 @@ This method is used to execute some tasks out of the **event loop** on a libuv
 worker thread. Subclasses must implement this method and the method is run on
 a thread other than that running the main event loop.  As the method is not
 running on the main event loop, it must avoid calling any methods from node-addon-api
-or running any code that might invoke JavaScript.  Instead once this method is
+or running any code that might invoke JavaScript. Instead once this method is
```
Comment (gabrielschulhof) [inline doc/async_worker.md]: nit: comma between "Instead" and "once"

File: doc/async_worker.md [hunk 2]
```diff
@@ -109,29 +109,29 @@ calls the callback provided when the AsyncWorker class was created, passing
...
-explicit AsyncWorker(const Function& callback);
+explicit Napi::AsyncWorker(const Function& callback);
 ```
 
 - `[in] callback`: The function which will be called when an asynchronous
 operations ends. The given function is called from the main event loop thread.
 
-Returns an AsyncWork instance which can later be queued for execution by calling
+Returns a`Napi::AsyncWork` instance which can later be queued for execution by calling
 `Queue`.
```
Comment (gabrielschulhof) [inline doc/async_worker.md]: nit: &#96;Queue&#96; should be fully namespaced.

File: doc/async_worker.md [hunk 3]
```diff
@@ -140,16 +140,15 @@ operations ends. The given function is called from the main event loop thread.
 identifier for the kind of resource that is being provided for diagnostic
 information exposed by the async_hooks API.
 
-Returns an AsyncWork instance which can later be queued for execution by calling
+Returns a `Napi::AsyncWork` instance which can later be queued for execution by calling
 `Queue`.
```
Comment (gabrielschulhof) [inline doc/async_worker.md]: nit: &#96;Queue&#96; should be fully namespaced.

File: doc/async_worker.md [hunk 4]
```diff
@@ -160,31 +159,30 @@ information exposed by the async_hooks API.
 - `[in] resource`: Object associated with the asynchronous operation that
 will be passed to possible async_hooks.
 
-Returns an AsyncWork instance which can later be queued for execution by calling
+Returns a `Napi::AsyncWork` instance which can later be queued for execution by calling
 `Queue`.
```
Comment (gabrielschulhof) [inline doc/async_worker.md]: nit: &#96;Queue&#96; should be fully namespaced.

File: doc/async_worker.md [hunk 5]
```diff
@@ -160,31 +159,30 @@ information exposed by the async_hooks API.
...
+explicit Napi::AsyncWorker(const Object& receiver, const Function& callback);
 ```
 
 - `[in] receiver`: The `this` object passed to the called function.
 - `[in] callback`: The function which will be called when an asynchronous
 operations ends. The given function is called from the main event loop thread.
 
-Returns an AsyncWork instance which can later be queued for execution by calling
+Returns a `Napi::AsyncWork` instance which can later be queued for execution by calling
 `Queue`.
```
Comment (gabrielschulhof) [inline doc/async_worker.md]: nit: &#96;Queue&#96; should be fully namespaced.

File: doc/async_worker.md [hunk 6]
```diff
@@ -160,31 +159,30 @@ information exposed by the async_hooks API.
...
 
-
 ### Constructor
 
-Creates a new `AsyncWorker`.
+Creates a new `Napi::AsyncWorker`.
 
 ```cpp
-explicit AsyncWorker(const Object& receiver, const Function& callback,const char* resource_name);
+explicit Napi::AsyncWorker(const Object& receiver, const Function& callback,const char* resource_name);
```
Comment (gabrielschulhof) [inline doc/async_worker.md]: nit: The types of the formal parameters should probably be fully namespaced.

File: doc/async_worker.md [hunk 7]
```diff
@@ -215,15 +212,15 @@ information exposed by the async_hooks API.
 - `[in] resource`: Object associated with the asynchronous operation that
 will be passed to possible async_hooks.
 
-Returns an AsyncWork instance which can later be queued for execution by calling
+Returns a `Napi::AsyncWork` instance which can later be queued for execution by calling
 `Queue`.
```
Comment (gabrielschulhof) [inline doc/async_worker.md]: nit: &#96;Queue&#96; should be fully namespaced.

File: doc/async_worker.md [hunk 8]
```diff
@@ -194,16 +192,15 @@ operations ends. The given function is called from the main event loop thread.
...
 
-
 ### Constructor
 
-Creates a new `AsyncWorker`.
+Creates a new `Napi::AsyncWorker`.
 
 ```cpp
-explicit AsyncWorker(const Object& receiver, const Function& callback, const char* resource_name, const Object& resource);
+explicit Napi::AsyncWorker(const Object& receiver, const Function& callback, const char* resource_name, const Object& resource);
```
Comment (gabrielschulhof) [inline doc/async_worker.md]: nit: The types of the formal arguments should probably be namespaced as well.

File: doc/async_worker.md [hunk 9]
```diff
@@ -232,25 +229,25 @@ virtual ~AsyncWorker();
...
-used to mix usage of the C N-API and node-addon-api.
+Returns the N-API napi_async_work wrapped by the `Napi::AsyncWorker` object. This
+can be used to mix usage of the C N-API and node-addon-api.
 
 ## Example
 
-The first step to use the `AsyncWorker` class is to create a new class that inherit
-from it and implement the `Execute` abstract method. Typically input to your
+The first step to use the `Napi::AsyncWorker` class is to create a new class that
+inherits from it and implement the `Execute` abstract method. Typically input to your
```
Comment (gabrielschulhof) [inline doc/async_worker.md]: nit: full namespacing for &#96;Execute&#96;

File: doc/async_worker.md [hunk 10]
```diff
@@ -232,25 +229,25 @@ virtual ~AsyncWorker();
...
 worker will be saved within class' fields generally passed in through its
 constructor.
 
 When the `Execute` method completes without errors the `OnOK` function callback
```
Comment (gabrielschulhof) [inline doc/async_worker.md]: nit: full namespacing for &#96;Execute&#96; and &#96;OnOK&#96;. Same with instances below.

File: doc/string.md [hunk 11]
```diff
@@ -45,21 +45,21 @@ Returns a UTF-16 encoded C++ string.
...
+Returns a new empty `Napi::String`.
 
 ### New
 ```cpp
-String::New(napi_env env, const std::string& value);
-String::New(napi_env env, const std::u16string& value);
-String::New(napi_env env, const char* value);
-String::New(napi_env env, const char16_t* value);
+Napi::String::New(napi_env env, const std::Napi::string& value);
+Napi::String::New(napi_env env, const std::u16Napi::string& value);
```
Comment (mhdawson) [inline doc/string.md]: I think these 2 might not be right as I don't expect std::Napi

Comment (NickNaso) [reply doc/string.md]: Ok just fixed!

**Comment by NickNaso** [comment]
@mhdawson  I'm working on `Napi::AsyncWorker` now to fix the probleme reported by @gabrielschulhof.

File: doc/async_worker.md (hunk 1 above)
Comment (NickNaso) [reply doc/async_worker.md]: Done

File: doc/async_worker.md (hunk 2 above)
Comment (NickNaso) [reply doc/async_worker.md]: Done

File: doc/async_worker.md (hunk 3 above)
Comment (NickNaso) [reply doc/async_worker.md]: Done

File: doc/async_worker.md (hunk 4 above)
Comment (NickNaso) [reply doc/async_worker.md]: Done

File: doc/async_worker.md (hunk 5 above)
Comment (NickNaso) [reply doc/async_worker.md]: Done

File: doc/async_worker.md (hunk 6 above)
Comment (NickNaso) [reply doc/async_worker.md]: Done

File: doc/async_worker.md (hunk 7 above)
Comment (NickNaso) [reply doc/async_worker.md]: Done

File: doc/async_worker.md (hunk 8 above)
Comment (NickNaso) [reply doc/async_worker.md]: Done

File: doc/async_worker.md (hunk 9 above)
Comment (NickNaso) [reply doc/async_worker.md]: Done

File: doc/async_worker.md (hunk 10 above)
Comment (NickNaso) [reply doc/async_worker.md]: Done

**Comment by NickNaso** [comment]
@mhdawson @gabrielschulhof  Ended to fix reported problems

**Comment by mhdawson** [comment]
Landed as fc11c944b2e1e828a9baadfc3807a6d9c52535ba.  Thanks for all of the hard work!

**Pull request closed without merging**